var/spack/repos/builtin/packages/amrex/package.py

```diff
@@ -68,20 +68,23 @@ class Amrex(CMakePackage, CudaPackage, ROCmPackage):
             description='Build particle classes')
     variant('plotfile_tools', default=False,
             description='Build plotfile_tools like fcompare')
-    variant('sundials', default=False,
-            description='Build AMReX with SUNDIALS support')
     variant('hdf5',  default=False,
             description='Enable HDF5-based I/O')
     variant('hypre', default=False,
             description='Enable Hypre interfaces')
     variant('petsc', default=False,
             description='Enable PETSc interfaces')
+    variant('sundials', default=False,
+            description='Enable SUNDIALS interfaces')
     variant('pic', default=False,
             description='Enable PIC')
 
     # Build dependencies
     depends_on('mpi', when='+mpi')
-    depends_on('sundials@4.0.0:4.1.0 +ARKODE +CVODE', when='@19.08: +sundials')
+    depends_on('sundials@4.0.0:4.1.0 +ARKODE +CVODE', when='@19.08:20.11 +sundials')
+    depends_on('sundials@5.7.0: +ARKODE +CVODE', when='@21.07: +sundials')
+    for arch in CudaPackage.cuda_arch_values:
+        depends_on('sundials@5.7.0: +ARKODE +CVODE +cuda cuda_arch=%s' % arch, when='@21.07: +sundials +cuda cuda_arch=%s' % arch)
```

Comment:
@balos1 What is the expected outcome when doing `spack install amrex~mpi`? I expect it should link to `sundials~mpi` correct? We might want to add extra depend rules (something like below)

https://github.com/spack/spack/blob/c6c9213766c5755ae893d202d1336883ad2296fa/var/spack/repos/builtin/packages/amr-wind/package.py#L68-L72

Reply:
Currently, the AMReX interfaces to SUNDIALS do not involve MPI, so any combination of `amrex+mpi` and `amrex~mpi` with `sundials+mpi` or `sundials~mpi` should be fine (I just tested them to confirm).

Reply:
We do want to add similar support for `+rocm` like `+cuda`.

Reply:
That would be great!  Do you want to do it in this PR or a follow-up one?

Reply:
I think we should try and include it in this PR. I am attempting to build amrex +rocm on Spock (so far unsuccessful), will follow up with more details on slack.

Reply:
OK, I pushed up support for `+rocm`  and verified it builds on spock. I think this PR should be good to go now.